pandas/core/exchange/column.py

```diff
@@ -0,0 +1,368 @@
+from functools import cached_property
```

Comment:
how does this compare to cache_readonly?

Reply:
That's a very good question which made me read much better about `cached_property` (and I wasn't aware `cache_readonly` existed already), thanks!

`cached_property` should (theoretically) work a little faster on non-first access as it assigns the attribute to the object's `__dict__`, but the attribute is writeable which isn't what we want here, so I'm changing it to `cache_readonly` instead.